### PR TITLE
external/stdk:update makefile of stdk and example

### PR DESCRIPTION
**Pull request by SRCN-LiQin**
update makefile to avoid build issue when make clean or distclean for non stdk case.

make[3]: Entering directory '/root/tizenrt/apps/examples/stdk_smart_lamp'
cp: cannot stat '/root/tizenrt/os/../external/stdk/st-device-sdk-c/src/include/profile/json/device_info_example.json': No such file or directory
xxd: /device_info.json: No such file or directory

make[2]: Entering directory '/root/tizenrt/external/stdk'
/bin/sh: 1: cannot create ./st-device-sdk-c/src/iot_root_ca.c: Directory nonexistent
sed: can't read ./st-device-sdk-c/src/iot_root_ca.c: No such file or directory
rm: cannot remove './st-device-sdk-c/src/iot_root_ca.c.bak': No such file or directory

@sunghan-chang

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [bb4e4f0fd9ff20a0838fa21e0d4d7e2d23ce284f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [bb4e4f0fd9ff20a0838fa21e0d4d7e2d23ce284f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1cba07f2bb6ae3e9cae615e121ae648ceb1e4dcc] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [1cba07f2bb6ae3e9cae615e121ae648ceb1e4dcc] - Code Rule Check (C++) OK.

File: apps/examples/stdk_smart_lamp/Makefile [hunk 1]
```diff
@@ -167,6 +138,32 @@ $(BUILTIN_REGISTRY)$(DELIM)$(FUNCNAME).bdat: $(DEPCONFIG) Makefile
 	$(Q) $(call REGISTER,$(APPNAME),$(FUNCNAME),$(THREADEXEC),$(PRIORITY),$(STACKSIZE))
 
 context: $(BUILTIN_REGISTRY)$(DELIM)$(FUNCNAME).bdat
+####################################
```
Comment (sunghan-chang) [inline apps/examples/stdk_smart_lamp/Makefile]: To be commands in a tab, a line should be started with a tab.
https://www.gnu.org/software/make/manual/html_node/Rule-Syntax.html#Rule-Syntax
```
The recipe lines start with a tab character
```

Is this called when context is executed?

**Comment by seinfra** [comment]
Target : [bccd65fd2afed42927816d1e8b6e8a4392eb6f21] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [bccd65fd2afed42927816d1e8b6e8a4392eb6f21] - Code Rule Check (C++) OK.

**Comment by sunghan-chang** [comment]
@SRCN-LiQin Could you let me know why you don't use context tab?

**Comment by seinfra** [comment]
Target : [0163ab4a8ca4121180aa602b94f6ed03f9280833] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0163ab4a8ca4121180aa602b94f6ed03f9280833] - Code Rule Check (C++) OK.

**Comment by SRCN-LiQin** [comment]
> @SRCN-LiQin Could you let me know why you don't use context tab?

updated with latest commit, thanks for your suggestion, plz check it.

File: external/stdk/Makefile [hunk 2]
```diff
@@ -131,7 +123,19 @@ $(COBJS): %$(OBJEXT): %.c
 	$(call ARCHIVE, $(BIN), $(OBJS))
 	$(Q) touch .built
 
-.depend: Makefile $(SRCS)
+#################################################
+# certificate
+#################################################
+context:
```
Comment (sunghan-chang) [inline external/stdk/Makefile]: Makefile does not execute the context target in the external folder. If you need this, you should add the folder path in Directories.mk.
Could you check an example as shown below?
```
ifeq ($(CONFIG_ENABLE_IOTIVITY),y)
CONTEXTDIRS += $(EXTDIR)$(DELIM)iotivity
endif
```

**Comment by sunghan-chang** [comment]
@SRCN-LiQin Others are good.

File: external/stdk/Makefile (hunk 2 above)
Comment (SRCN-LiQin) [reply external/stdk/Makefile]: I agree that this solution is more graceful, thanks for suggestion. updated.

**Comment by seinfra** [comment]
Target : [267c71417226fb7bc087f014f14151c1b6b1e743] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [267c71417226fb7bc087f014f14151c1b6b1e743] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b89a534f026323a37f98a04a7128dd9b173a51c8] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b89a534f026323a37f98a04a7128dd9b173a51c8] - Code Rule Check (C++) OK.

**Pull request merged**